STO doc AIDA remediation advice DOC-3156 (#1899)
harness/developer-hub: docs/security-testing-orchestration/use-sto/view-and-troubleshoot-vulnerabilities/ai-based-remediations.md

---
title: Fix issues using AI-enhanced remediation steps
description: Enhanced remediation using Harness AI Development Assistant (AIDA)
sidebar_position: 35
---

<!-- style>

.green {
    color: green;
    font-weight:700;
    font-size: 30px;
}

.hidden {
   display: none;
}
</style -->

```mdx-code-block
import remediation_issue from '../static/ai-remediation-issue.png'
import remediate_indiv_occurrences from '../static/ai-remediation-occurrences.png'
import remediate_occurrence_three from '../static/ai-remediation-occurrence-3.png'
import remediate_issue_refine from '../static/ai-remediation-issue-refine.png'
```

> Harness AI Development Assistant (AIDA) auto-generates remediation advice for detected vulnerabilities. Harness AIDA explains the vulnerability, gives solutions how to fix it, and (where applicable) provides updated code snippets. 

Reviewer: ```suggestion
Harness AI Development Assistant (AIDA) auto-generates remediation advice for detected vulnerabilities. Harness AIDA explains the vulnerability, provides suggestions on how to fix it, and (where applicable) provides updated code snippets. 
```

   This suggested remediation is based on public information about the CVE or CWE and the first detected occurrence (**Occurrence 1**) in the target. If the scanner captures the specific line where the vulnerability is occurring, the query to Harness AIDA includes this line as well.    

      ```mdx-code-block
      <img src={remediation_issue} alt="Generate a new remediation using additional information." height="50%" width="50%" />
      ```

3. If you're not satisfied with the initial advice, click **Regenerate**.  

2. If you want to regenerate the advice with additional information or context, do the following: 

   1. click **Edit Input**. 

   2. Specify the occurrence, reference ID, and language (if you've scanned a codebase). 
   
     Harness AIDA can often auto-detect the language of a code snippet, but it's good practice to confirm that the language setting is correct. 

   3. Add any additional context in the text pane. This might be a code snippet, a Dockerfile, or some other piece of information that's relevant to the specific target and issue. Then click **Generate**. 

>       :::note

Reviewer: ```suggestion
      :::info note
```

   
   An issue might have no known remediation, especially if it was recently discovered. An issue might have multiple suggested remediations that are contradictory or applicable only to specific use cases. 

* The workflow description below shows how you can refine a suggestion by providing more information, such as additional context or code snippets, to Harness AIDA.

## Workflow description

1. When you go to **Security Tests** and then select an issue, an initial **AI enhanced remediation** appears in **Issue Details**. 

   This suggested remediation is based on public information about the CVE or CWE and the first detected occurrence (**Occurrence 1**) in the target. If the scanner captures the specific line where the vulnerability is occurring, the query to Harness AIDA includes this line as well.    

      ```mdx-code-block
      <img src={remediation_issue} alt="Generate a new remediation using additional information." height="50%" width="50%" />
      ```

3. If you're not satisfied with the initial advice, click **Regenerate**.  

2. If you want to regenerate the advice with additional information or context, do the following: 

>    1. click **Edit Input**. 

Reviewer 2: ```suggestion
   1. Select **Edit Input**. 
```
